src/Altinn.App.Core/Helpers/ObjectUtils.cs

```diff
@@ -137,14 +138,31 @@ public static void PrepareModelForXmlStorage(object model, int depth = 64)
                 SetToDefaultIfShouldSerializeFalse(model, prop, methodInfos);
 
                 // Set string properties with [XmlText] attribute to null if they are empty or whitespace
-                if (
-                    value is string s
-                    && string.IsNullOrWhiteSpace(s)
-                    && prop.GetCustomAttribute<XmlTextAttribute>() is not null
-                )
+                if (value is string s)
                 {
-                    // Ensure empty strings are set to null
-                    prop.SetValue(model, null);
+                    if (string.IsNullOrWhiteSpace(s))
+                    {
+                        if (prop.GetCustomAttribute<XmlTextAttribute>() is not null)
+                        {
+                            // Ensure empty strings are set to null
+                            prop.SetValue(model, null);
+                        }
+                    }
+                    else
+                    {
+                        var xmlRegex = XmlInvalidCharsRegex();
+                        if (prop.SetMethod is not null)
+                        {
+                            // Remove invalid xml characters
+                            prop.SetValue(model, xmlRegex.Replace(s, "�"));
+                        }
+                        else if (xmlRegex.IsMatch(s))
+                        {
+                            throw new InvalidOperationException(
+                                $"Property {prop.Name} of datamodel contains invalid xml characters"
```

Comment:
This looks wrong. This error shows up if
1. value is string
2. not string.IsNullOrWhitespace
3. prop.SetMethod is null
4. xmlRegex.IsMatch(s)

Especially the 3 seems unlikely to be true, and the message does not make sense in that case.

Reply:
I didn't have this branch orignally, but ran into it because some of the models have `public string Prop => ""` kind of properties

Reply:
I don't understand what branch?

I'm trying to understand when you want to throw exception and when you want to just replace the offending characters,

Reply:
It replaces when it can (the property _isnt_ readonly), and throws error when it can't (the property is readonly)

Reply:
Now it makes sense.

I would just assume that the readonly fields doesn't take user updates (they can't be set via json deserialisation), so we don't need to inspect them. If we do conclude that it makes sense to validate readonly this is a sensible approach.

Reply:
yeah it's not strictly necessary, we will just get a similar error from XmlSerializer if it ends up getting serialized (and has invalid chars)

Reply:
To be strictly correct you would need to check for `[XmlIgnore]`. But I should probably not suggest more branches in order suggest removing a branch.